src/server/main_service.cc

```diff
@@ -409,7 +409,8 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
   VLOG(2) << "Got: " << args;
 
   string_view cmd_str = ArgS(args, 0);
-  bool is_trans_cmd = (cmd_str == "EXEC" || cmd_str == "MULTI" || cmd_str == "DISCARD");
+  bool is_trans_cmd = (cmd_str == "EXEC" || cmd_str == "MULTI" || cmd_str == "DISCARD"
+    || cmd_str == "WATCH");
```

Comment:
why `WATCH` is here?

Reply:
Because it doesn't have to be recorded and we either:
* add an exception like with `SELECT` that returns an error
* let it respond on its own

On a second thought, the first approach probably seems more consistent